lib/chef/provider/user.rb

```diff
@@ -95,6 +107,11 @@ def define_resource_requirements
           a.whyrun "ruby-shadow is not installed. Attempts to set user password will cause failure.  Assuming that this gem will have been previously installed." \
             "Note that user update converge may report false-positive on the basis of mismatched password. "
         end
+        requirements.assert(:all_actions) do |a|
+          a.assertion { !((new_resource.expire_date || new_resource.inactive) && platform_family?(%w{aix freebsd mac_os_x netbsd openbsd openindiana solaris2 windows})) }
```

Comment:
don't blacklist, that's silly. Do a.assertion { ... || platform_family?('linux') }` - and I think there's actually `linux?` helper you can just do instead.

Reply:
how does this look now?